UITests/Pages/MainPage.cs

```diff
@@ -0,0 +1,83 @@
+﻿using OpenQA.Selenium;
+using OpenQA.Selenium.Support.UI;
+using SeleniumExtras.PageObjects;
+using UserUITest.Pages;
+
+namespace UserManagementServiceUITests.Pages
+{
+    public class MainPage : BasePage
+    {
+        [FindsBy(How = How.CssSelector, Using = ".content.px-4")]
+        private IWebElement _elementSection;
+
+        [FindsBy(How = How.CssSelector, Using = "[aria-label = 'press click to search']")]
+        private IWebElement _searchButton;
+
+        [FindsBy(How = How.XPath, Using = "//*[@inputmode ='text' ]")]
+        private IWebElement _searchField;
+
+        [FindsBy(How = How.XPath, Using = "//b")]
+        private IList<IWebElement> _productNames;
+
+        [FindsBy(How = How.XPath, Using = "//b/ancestor::h4/preceding-sibling::h4")]
+        private IList<IWebElement> _productManufactors;
+
+        [FindsBy(How = How.ClassName, Using = "mud-snackbar-content-message")]
+        private IWebElement _infoMessageWindow;
+
+        [FindsBy(How = How.ClassName, Using = "me-auto")]
+        private IWebElement _errorMessage;
+
+        public MainPage(IWebDriver driver) : base(driver)
+        {
+        }
+
+        public void WaitProductsLoading()
+        {
+            WebDriverWait wait = new WebDriverWait(_driver, TimeSpan.FromSeconds(50));
+            wait.Until((_) => _elementSection.Displayed);
+        }
+
+        public void WaitAmountOfExpectedProducts(int amount)
+        {
+            WebDriverWait wait = new WebDriverWait(_driver, TimeSpan.FromSeconds(50));
+            wait.Until((_) => _productNames.Count == amount);
```

Comment:
If the number of elements found is greater than the value passed in the amount variable, the test will fail. Maybe you should choose another condition as an expectation, and check the expected number of products, if necessary, in assertions?